docs/system_overview/configuration.md

```diff
@@ -14,254 +14,676 @@ This scrambled password file is saved after an `iinit` is run. If this file does
+        ]
+    },
+
+    // Holds the log level for various log message categories used throughout the server.
+    // Missing log categories default to "info".
+    // 
+    // The following values are supported: trace, debug, info, warn, error, critical
+    "log_level": {
+        "agent": "info",
+        "agent_factory": "info",
+        "api": "info",
+        "authentication": "info",
+        "database": "info",
+        "delay_server": "info",
+        "legacy": "info",
+        "microservice": "info",
+        "network": "info",
+        "resource": "info",
+        "rule_engine": "info",
+        "server": "info"
+    },
+
+    // Indicates to iRODS whether to use the hash used by the client or the data at rest, or to force
+    // the use of the default hash scheme: strict or compatible.
+    "match_hash_policy": "compatible",
+
+    // A 32-byte encryption key shared by the zone for use in the advanced negotiation handshake at the
+    // beginning of an iRODS client connection.
+    "negotiation_key": "",
+
+    // Holds several different properties such as database connection information, rule engine configuration, etc.
+    "plugin_configuration": {
+        // TODO
+        "authentication": {},
+
+        // (Optional)
+        // Defines database connection information.
+        // This only applies to servers running as a catalog provider or delay server.
+        "database": {
+            // The type of database iRODS is using for the catalog.
+            // The following property names are supported: postgres, mysql, and oracle
+            "catalog_database_type": {
+                // The hostname of the database server (can be localhost).
+                "db_host": "localhost",
+
+                // The name of the database used as the catalog.
+                "db_name": "ICAT",
+
+                // The name of the ODBC entry used by the server (normally defined in /etc/odbcinst.ini).
+                "db_odbc_driver": "",
+
+                // The password for the "db_username" to connect to the "db_name".
+                "db_password": "",
+
+                // The port number on which the database server is listening.
+                "db_port": 0,
+
+                // The database user name.
+                "db_username": ""
+            }
+        },
 
-  - `catalog_database_type` (required) - The type of database iRODS is using for the iCAT, either 'postgres', 'mysql', or 'oracle'
-  - `db_host` (required) - The hostname of the database server (can be localhost)
-  - `db_odbc_type` (required) - The ODBC type, usually 'unix'
-  - `db_password` (required) - The password for the `db_username` to connect to the `db_name`
-  - `db_port` (required) - The port on which the database server is listening
-  - `db_name` (required) - The name of the database used as the iCAT
-  - `db_username` (required) - The database user name
+        // TODO
+        "network": {},
+
+        // TODO
+        "resource": {},
```

Comment:
Needs a description.

When is this used? izonereport perhaps? Anything else?

Reply:
i think this is just for completeness - that each plugin type has a section.     i don't know that we use this section, since we have 'context string' per resource for most of this type of thing.

Reply:
Marked as "This property can be ignored."